src/senaite/core/browser/fields/records.py

```diff
@@ -147,16 +147,23 @@ def validate(self, value, instance, errors={}, **kwargs):
         if errors and errors.has_key(name):
             return True
 
-        result = None
-        for record in value:
-            result = RecordField.validate(self,
-                                          record,
-                                          instance,
-                                          errors={},
-                                          **kwargs
-            )
-            if result: return result
-        return result
+        validation_results = (
+            RecordField.validate(
+                self,
+                record,
+                instance,
+                errors={},
+                **kwargs
+                )
+            for record
+            in value
+        )
+        return next((
+            result
+            for result
+            in validation_results
+            if result is not None
+        ), __default=None)
```

Comment:
Admittedly these are now more lines of code but I find it easier to follow—especially because it has one return statement less.

Reply:
agree